src/LsmTree.hs

```diff
@@ -13,18 +13,27 @@ module LsmTree
        , initFileSystem
        , insertTree
        , insertTable
+         -- * foldr
+       , sizeTree
        ) where
 
 -- | Data type that stores values while program is working.
 data Tree a
-    = TreeNode (Tree a) a (Tree a)
+    = TreeNode (Tree a) !a (Tree a)
     | Leaf
     deriving stock (Eq, Show)
 
+instance Foldable Tree where
+    foldMap _ Leaf = mempty
+    foldMap f (TreeNode l x r) = foldMap f l <> f x <> foldMap f r
+
+sizeTree :: Tree a -> Int
+sizeTree = foldr (const (+1)) 0
```

Comment:
This implementation is not efficient. It's better to use `foldl'` -- a strict fold

Reply:
okey, I agree :D